### ISSUE TITLE
Probably missing StubChain#invocation_order entry for and_throw

### ISSUE THREAD

**Issue by tbuehlmann** [rspec/rspec-mocks#1067]
When calling

```rb
require 'rspec/mocks/standalone'
allow_any_instance_of(Object).to receive(:foo).and_throw(:bar)
```

I get:

```
NoMethodError: undefined method `include?' for nil:NilClass
from /home/tobias/.gem/ruby/2.3.0/gems/rspec-mocks-3.4.1/lib/rspec/mocks/any_instance/stub_chain.rb:41:in `verify_invocation_order'
```

When looking into [StubChain#verify_invocation_order](https://github.com/rspec/rspec-mocks/blob/v3.4.1/lib/rspec/mocks/any_instance/stub_chain.rb#L39-L41), it seems there's missing an entry for `and_throw` in the [invocation_order](https://github.com/rspec/rspec-mocks/blob/v3.4.1/lib/rspec/mocks/any_instance/stub_chain.rb#L28-L37) method:

```rb
def invocation_order
  @invocation_order ||= {
    :with => [nil],
    :and_return => [:with, nil],
    :and_raise => [:with, nil],
    :and_yield => [:with, nil],
    :and_call_original => [:with, nil],
    :and_wrap_original => [:with, nil]
  }
end

def verify_invocation_order(rspec_method_name, *_args, &_block)
  return if invocation_order[rspec_method_name].include?(last_message)
  raise NoMethodError, "Undefined method #{rspec_method_name}"
end
```

Amirite or am I missing something obvious here?

**Comment by myronmarston** [comment]
It looks like you are right.  Want to work up a fix?

**Comment by tbuehlmann** [comment]
Sure! Wrote something [up](https://github.com/rspec/rspec-mocks/pull/1068).

**Comment by JonRowe** [comment]
Closed by #1067
